nio/api.py

```diff
@@ -381,12 +382,22 @@ def login(
         path = Api._build_path(path=["login"])
 
         if password is not None:
+            identifier = {}
+            if '@' in user and ':' not in user:
```

Comment:
I think a better check would be to see if the ID starts with an `@` and if so, consider it a user id, otherwise an email address.

Reply:
I am not really familiar with Matrix. If you assume this is a sound check, I'll change this.
Cannot a user log in with, eg, `tobast` instead of `@tobast`?

Reply:
The grammar is described in the [spec](https://spec.matrix.org/unstable/appendices/#user-identifiers).

If users pass in only the local part it can't contain an `@`, if they pass in the whole user ID it must start with an `@`. The local part can't contain an `@` as per spec.

Reply:
Seems sound indeed.